platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/pipeline/VirtualPipeline.java

```diff
@@ -213,6 +219,28 @@ private void applyFlushBackpressure() {
         final Duration sleepTime = CompareTo.min(computedSleepTime, maxSleepTime);
 
         try {
+            logger.debug(VIRTUAL_MERKLE_STATS.getMarker(), "Flush backpressure: {} ms", sleepTime.toMillis());
+            MILLISECONDS.sleep(sleepTime.toMillis());
+        } catch (final InterruptedException ex) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    /**
+     * Slow down the fast copy operation if total size of all (unreleased) virtual root copies
+     * in this pipeline exceeds {@link VirtualMapSettings#getTotalFlushThreshold()}.
+     */
+    private void applyTotalSizeBackpressure() {
+        final long totalSize = currentTotalSize();
+        final long sizeThreshold = VirtualMapSettingsFactory.get().getTotalFlushThreshold();
+        final double ratio = (double) totalSize / sizeThreshold;
+        final int over100percentExcess = (int) ((ratio - 1.0) * 100);
+        if (over100percentExcess < 0) {
+            return;
+        }
+        final Duration sleepTime = Duration.ofMillis((long) over100percentExcess * over100percentExcess);
```

Comment:
This looks suspicious. I'd guess that you wanted to have something like `ratio * over100percentExcess ` here.

Reply:
It converts a double (ratio) to percents. If the result is greater than 100%, the overhead is squared and used as a delay.

A few examples:
* Assume the current total size is 2000, with default threshold is 2500. `ratio` is 0.8, `over100percentExcess` is -20 (which means -20 percent), no backressure
* If current total size is 3000, `ratio` is 1.2, `over100percentExcess` is 20, and backpressure delay is 20*20 = 400 (milliseconds)
* If current total size is double threshold (5Gb with default settings), the delay will be 10 seconds (100 percent squared = 10000ms = 10s)